### PR TITLE
Move Layer utility functions to /napari/layers/layer_utils.py

### PR DESCRIPTION
**Pull request by kevinyamauchi**
# Description
As discussed in #761, this PR moves the the util functions used for multiple layer types from /napari/util/misc.py to `napari/layers/layers/layer_utils.py `. I also added some tests for `increment_unnamed_colormap()` and `segment_normal()` and fixed a typo in the `increment_unnamed_colormap()` docstring.

## Type of change
- [x] Bug-fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

# References
This is the third of a series of PRs addressing #761 

# How has this been tested?
- [x] Added tests for `increment_unnamed_colormap()` and `segment_normal()` in `/napari/layers/tests/layer_utils_tests.py'
- [x] All layer model tests pass
- [x] Tested `/examples/add_image.py`

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by sofroniewn — APPROVED** [review]
Love it, another beautiful PR. Thanks for adding more tests. I will approve and then wait till tomorrow for any additional feedback before merge

**Comment by kevinyamauchi** [comment]
Sounds good!

**Pull request merged**
